### ISSUE TITLE
TS2377 not reported when 'super' is called conditionally

### ISSUE THREAD

**Issue by vostrnad** [microsoft/TypeScript#45377]
# Bug Report

The TS2377 error is reported when the constructor of a derived class does not contain a call to 'super'. However, it is not reported when only some branches contain the call but not others.

### 🔎 Search Terms

TS2377 super class missing branch

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about **Classes**

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDCCuEAuB7AtgUQE5ZV6ApgB5IEB2AJjNrvgN4BQ00wKZyW8wqWAFAG5gQ8AgC5oHAJZkA5tAA+0MvDQAjAlgCU0Rs2aSAZtF5IAngAcCKI4OEFoAXifQA5FNkvtuvcwjxLfLYimkx6AL6EIBD23j4A9HHQKCjmEAA0SigS-hq82sBCIACEocxhoeVhQA)

### 💻 Code

```ts
class CustomError extends Error {
  constructor(value: string | number) {
    if (typeof value === 'string') {
      super(value)
    } else {
      // oops, no super() call!
    }
  }
}
```

### 🙁 Actual behavior

TypeScript does not report any errors for the above code.

### 🙂 Expected behavior

TypeScript should report TS2377 for the above code, since 'super' isn't called in every branch.
